Source_Code/Common/Board.java

```diff
@@ -0,0 +1,150 @@
+package Common;
+import Common.Messages.ActionRequests.MoveRequest;
+
+/**
+ * This class defines the set up of the Clue-less board with some useful helper 
+ * classes so that other
+ * 
+ * @author Steve Nilla
+ */
+public class Board
+{
+  // set up board
+  BoardRoom board[][] = new BoardRoom[5][5];
+ 
+  /**
+   * The method to move a player on the board
+   *
+   * @param player
+   * @param move
+   */
+  public void movePlayer(Player player, MoveRequest move) throws IllegalArgumentException
+  {
+    // get current player position
+    int currX = player.getxPos();
+    int currY = player.getyPos();
+
+    // new player x,y positions
+    int newX;
+    int newY;
+
+    // set prospective player location
+    switch(move)
+    {
+      case MoveRequest.UP:
+        // toggle one space up
+        newX = currX;
+        newY = currY - 1;
+        break;
+      case MoveRequest.DOWN:
+        // toggle one space down
+        newX = currX;
+        newY = currY + 1;
+        break;
+      case MoveRequest.LEFT:
+        // toggle one space to the left
+        newX = currX - 1;
+        newY = currY;
+        break;
+      case MoveRequest.RIGHT:
+        // toggle one space to the right
+        newX = currX + 1;
+        newY = currY;
+        break;
+      case MoveRequest.STAY:
+        // do nothing
+        newX = currX;
+        newY = currY;
+        break;
+      case MoveRequest.SHORTCUT:
+        // move player diagonally across the board when they are in a corner
+        if ((currX == 0) && (currY == 0))
+        {
+          newX = 4;
+          newY = 4;
+        }
+        else if ((currX == 4) && (currY == 4))
+        {
+          newX = 0;
+          newY = 0;
+        }
+        else if ((currX == 4) && (currY == 0))
+        {
+          newX = 0;
+          newY = 4;
+        }  
+        else if ((currX == 0) && (currY == 4))
+        {
+          newX = 4;
+          newY = 0;
+        }  
+        else
+        {
+          throw new IllegalArgumentException("Invalid move request");
+        }
+        break;
+    }
+
+    // if room is an occupied hallway, player cannot move there
+    if (board[currX][currY] instanceof BoardHallway)
+    {
+      if (board[currX][currY].isOccupied())
```

Comment:
Again with up-tight settings. It won't allow implicit casting, so I had to change this to "if (((BoardHallway)board[currX][currY]).isOccupied())"